src/types/Auth.ts

```diff
@@ -0,0 +1,35 @@
+export interface AuthMethod {
+  method: 'METHOD_TOKEN' | 'METHOD_OIDC';
+  enabled: boolean;
+  sessionCompatible: boolean;
+  metadata: { [key: string]: any };
+}
+
+interface AuthMethodOIDCMetadataProvider {
+  authorize_url: string;
+  callback_url: string;
+}
+
+export interface AuthMethodOIDC extends AuthMethod {
+  method: 'METHOD_OIDC';
+  metadata: {
+    providers: Record<string, AuthMethodOIDCMetadataProvider>;
+  };
+}
+
+export interface AuthMethodOIDCMetadata {
+  'io.flipt.auth.oidc.email': string;
+  'io.flipt.auth.oidc.email_verified': string;
+  'io.flipt.auth.oidc.name': string;
+  'io.flipt.auth.oidc.picture'?: string;
+  'io.flipt.auth.oidc.provider': string;
```

Comment:
Would be good to explore this with different requested `scopes`.
I see `picture` is required. However, I think technically they all could be required.
Since you can configure Flipt with no scopes (apart from the `oidc` scope which is required) and we can authenticate them but obtain next to no information.

Which, I think, is good. It is a feature the operator can decide on.